device_test.go

```diff
@@ -111,4 +126,29 @@ func TestClaimAndRelease(t *testing.T) {
 	if err := dev.Close(); err != nil {
 		t.Fatalf("%s.Close(): got error %v, want nil", dev, err)
 	}
+
+	if _, err := dev.Config(cfgNum); err == nil {
```

Comment:
add a separate TestAutoDetach function - there's no need to attach it to the existing one.

You can still keep the SetAutoDetach(true) in existing function, because the default fakelibusb will happily return "nil" on detach, so no error an all is good.

And you can still keep the "SetAutoDetach fails on closed device" in the TestClaimAndRelease, that makes sense.